lib/fluent/plugin/in_tail.rb

```diff
@@ -911,16 +1000,35 @@ def opened?
         def handle_notify
           with_io do |io|
             begin
+              number_bytes_read = 0
+              start_reading = Fluent::Clock.now
               read_more = false
 
               if !io.nil? && @lines.empty?
                 begin
                   while true
-                    @fifo << io.readpartial(8192, @iobuf)
+                    @fifo << io.readpartial(BYTES_TO_READ, @iobuf)
                     @fifo.read_lines(@lines)
-                    if @lines.size >= @read_lines_limit
+
+                    number_bytes_read += BYTES_TO_READ
+                    limit_bytes_per_second_reached = (number_bytes_read >= @read_bytes_limit_per_second && @read_bytes_limit_per_second > 0)
+                    @log.debug("reading file: #{@path}")
+                    if @lines.size >= @read_lines_limit || limit_bytes_per_second_reached
                       # not to use too much memory in case the file is very large
                       read_more = true
+
+                      if limit_bytes_per_second_reached
+                        # sleep to stop reading files when we reach the read bytes per second limit, to throttle the log ingestion
+                        time_spent_reading = Fluent::EventTime.now - start_reading
```

Comment:
`Fluent::Clock.now`

Reply:
Oops! Fixed in https://github.com/fluent/fluentd/pull/3185/commits/dba436545387b426583d9404fa1fddb9659708b6.